Sources/ResilientDecoding/ResilientDictionary+DecodingOutcome.swift

```diff
@@ -0,0 +1,114 @@
+// Created by George Leontiev on 4/23/20.
+// Copyright © 2020 Airbnb Inc. All rights reserved.
+
+import Foundation
+
+extension Resilient {
+
+  init<T>(_ results: [String: Result<T, Error>]) where Value == [String: T] {
+    self.init(results, transform: { $0 })
+  }
+
+  init<T>(_ results: [String: Result<T, Error>]) where Value == [String: T]? {
+    self.init(results, transform: { $0 })
+  }
+
+  /**
+   - parameter transform: While the two lines above both say `{ $0 }` they are actually different because the first one is of type `([String: T]) -> [String: T]` and the second is of type `([String: T]) -> [String: T]?`.
+   */
+  private init<T>(_ results: [String: Result<T, Error>], transform: ([String: T]) -> Value) {
+    let elements = results.compactMapValues { try? $0.get() }
+    let value = transform(elements)
+    if elements.count == results.count {
+      self.init(value, outcome: .decodedSuccessfully)
+    } else {
+      #if DEBUG
+      let error = ResilientDecodingOutcome.DictionaryDecodingError(topLevelError: nil, results: results)
+      /// `DictionaryDecodingError` is not reported
+      self.init(value, outcome: .recoveredFrom(error, wasReported: false))
+      #else
+      self.init(value, outcome: .recoveredFromDebugOnlyError)
+      #endif
+    }
+  }
+
+}
+
+#if DEBUG
+
+extension ResilientDecodingOutcome {
+
+  /**
+   A type representing some number of errors encountered while decoding a dictionary
+   */
+  public struct DictionaryDecodingError<Element>: Error {
+    public let results: [String: Result<Element, Error>]
+    public var errors: [Error] {
+      /// It is currently impossible to have both a `topLevelError` and `results` at the same time, but this code is simpler than having an `enum` nested in this type.
```

Comment:
Yeah, I tried the enum and there is a substantial amount of boilerplate relative to the functionality (since the `enum` would nee to be private to `DictionaryDecodingError` to preserve the public API). I think the comment works well enough here.